### PR TITLE
Feature/per gene

### PR DESCRIPTION
**Pull request by henrikstranneheim**
### This PR fixes:

- Refactoring of addition to feature file output and better syntax to parse_vep_csq

### How to test:

- Automatic and continuous test pass

### Expected outcome:
- Installation, unit and integration tests pass

### Review:
- [x] Code review
- [x] New code is executed and covered by tests
- [ ] Tests pass

### CODE REVIEW

File: vcfparser.pl [hunk 1]
```diff
@@ -849,77 +849,69 @@ sub parse_vep_csq {
...
 
+      TRANSCRIPT:
         foreach my $transcript (@transcripts) {
 
-            my @transcript_effects = split( /\|/, $transcript );    #Split in "|"
-
-            ## Alias hgnc_id column number
-            my $hgnc_id_column_ref = \$vep_format_field_column_href->{HGNC_ID};
+            ## Split transcript into VEP CSQ fields
+            my @transcript_effects = split( /\|/, $transcript );
```
Comment (jemten) [inline vcfparser.pl]:
```suggestion
            my @transcript_effects = split /\|/, $transcript;
```

File: lib/MIP/File/Format/Vcf.pm [hunk 2]
```diff
@@ -171,6 +172,64 @@ sub convert_to_range {
     return $final_stop_position;
 }
 
+sub get_transcript_effects {
```
Comment (jemten) [inline lib/MIP/File/Format/Vcf.pm]: perltidy

File: vcfparser.pl [hunk 3]
```diff
@@ -849,77 +849,69 @@ sub parse_vep_csq {
...
 
-            ## Alias hgnc_symbol column number
-            my $hgnc_symbol_column_ref = \$vep_format_field_column_href->{SYMBOL};
+	    my %transcript_csq = get_transcript_effects({transcript_effects_ref => \@transcript_effects,
+vep_format_field_column_href => $vep_format_field_column_href});
```
Comment (jemten) [inline vcfparser.pl]: indentation

File: vcfparser.pl [hunk 4]
```diff
@@ -849,77 +849,69 @@ sub parse_vep_csq {
...
+                next TRANSCRIPT if ($per_gene);
 
-                        push( @{ $record_href->{select_transcripts} }, $transcript )
-                          ;    #Add all transcripts to selected transcripts
+                add_transcript_to_feature_file(
+                    {
+                        hgnc_id          => $transcript_csq{hgnc_id},
+                        vcf_record_href  => $record_href,
+                        select_data_href => $select_data_href,
+                        transcript       => $transcript,
```
Comment (jemten) [inline vcfparser.pl]: order AB

File: vcfparser.pl (hunk 1 above)
Comment (henrikstranneheim) [reply vcfparser.pl]: Will fix later

**Pull request merged**